pkg/rtc/types/interfaces.go

```diff
@@ -32,86 +32,110 @@ const (
-	// returns list of participant identities that the current participant is subscribed to
-	GetSubscribedParticipants() []livekit.ParticipantID
+	SubscriptionPermissionUpdate(publisherID livekit.ParticipantID, trackID livekit.TrackID, allowed bool)
+
+	// callbacks
+	OnStateChange(func(p Participant, oldState livekit.ParticipantInfo_State))
+	// OnTrackUpdated - one of its publishedTracks changed in status
+	OnTrackUpdated(callback func(Participant, PublishedTrack))
+	OnMetadataUpdate(callback func(Participant))
+	OnDataPacket(callback func(Participant, *livekit.DataPacket))
+	OnClose(_callback func(Participant, map[livekit.TrackID]livekit.ParticipantID))
+
+	// updates from remotes
+	UpdateSubscribedQuality(nodeID string, trackID livekit.TrackID, maxQuality livekit.VideoQuality) error
+	UpdateMediaLoss(nodeID string, trackID livekit.TrackID, fractionalLoss uint32) error
+
+	// session migration
+	SetMigrateState(s MigrateState)
+	MigrateState() MigrateState
+	AddMigratedTrack(cid string, ti *livekit.TrackInfo)
+	SetPreviousAnswer(previous *webrtc.SessionDescription)
+}
+
+//counterfeiter:generate . Participant
+type Participant interface {
+	ID() livekit.ParticipantID
+	Identity() livekit.ParticipantIdentity
+
+	ToProto() *livekit.ParticipantInfo
+
+	SetMetadata(metadata string)
+
+	GetPublishedTrack(sid livekit.TrackID) PublishedTrack
+	GetPublishedTracks() []PublishedTrack
+
+	AddSubscriber(op Participant, params AddSubscriberParams) (int, error)
+	RemoveSubscriber(op Participant, trackID livekit.TrackID)
 
 	// permissions
-	CanPublish() bool
-	CanSubscribe() bool
-	CanPublishData() bool
 	Hidden() bool
-	IsRecorder() bool
-	SubscriberAsPrimary() bool
 
 	Start()
 	Close() error
 
 	// callbacks
-	OnStateChange(func(p Participant, oldState livekit.ParticipantInfo_State))
 	// OnTrackPublished - remote added a remoteTrack
 	OnTrackPublished(func(Participant, PublishedTrack))
-	// OnTrackUpdated - one of its publishedTracks changed in status
-	OnTrackUpdated(callback func(Participant, PublishedTrack))
-	OnMetadataUpdate(callback func(Participant))
-	OnDataPacket(callback func(Participant, *livekit.DataPacket))
-	OnClose(func(Participant, map[livekit.TrackID]livekit.ParticipantID))
-
-	// package methods
-	AddSubscribedTrack(st SubscribedTrack)
-	RemoveSubscribedTrack(st SubscribedTrack)
-	SubscriberPC() *webrtc.PeerConnection
 
 	UpdateSubscriptionPermissions(permissions *livekit.UpdateSubscriptionPermissions, resolver func(participantID livekit.ParticipantID) Participant) error
-	SubscriptionPermissionUpdate(publisherID livekit.ParticipantID, trackID livekit.TrackID, allowed bool)
-
 	UpdateVideoLayers(updateVideoLayers *livekit.UpdateVideoLayers) error
 
-	UpdateSubscribedQuality(nodeID string, trackID livekit.TrackID, maxQuality livekit.VideoQuality) error
-
-	UpdateMediaLoss(nodeID string, trackID livekit.TrackID, fractionalLoss uint32) error
-
 	DebugInfo() map[string]interface{}
+
+	LocalParticipant
```

Comment:
maybe better to put this at beginning of  structure

Reply:
Will do